app/models/search/token.rb

```diff
@@ -106,36 +110,48 @@ def comparison(qual, op, rhs_or_values)
       end
     end
 
-    # generates an sql where clause fragment for a comparison with the given qualifier, operator, and value token
-    def comparison_fragment(qual, op, value_token)
+    def qualifier_has_more_than_one_column(qualifier)
+      qualifier.col.is_a? Array
+    end
+
+    def comparisons_for_all_columns(qualifier, op, lex_tok)
+      sql_conditions = qualifier.col.map do |column|
+                         comparison_fragment(qualifier, column, op, lex_tok)
+                       end
+
+      sql_conditions.join(' OR ')
```

Comment:
If you use a one letter loop var I think this could be a one-liner without loss of readability: 
`qualifier.col.map{ |c| comparison_fragment(qualifier, c, op, lex_tok) }.join(' OR ')`.

Please prefer code like this when available.